File: 83-RemoveDuplicatesFromSortedList-1.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        ListNode* dummy=head;
        while (head) {
            ListNode* node=head;
            while (node->next&&node->val==node->next->val) {
                node=node->next;
            }
            if (head!=node) head->next=node->next;
            head=head->next;
        }
        return dummy;
    }
};

File: 83-RemoveDuplicatesFromSortedList-1.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
public class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        ListNode dummy=head;
        while (head!=null) {
            ListNode node=head;
            int val=node.val;
            while (node.next!=null&&node.next.val==val) {
                node=node.next;
            }
            if (node!=head) {
                head.next=node.next;
            }
            head=head.next;
        }
        return dummy;
    }
}
